src/main/java/com/minecolonies/coremod/colony/managers/StatisticsManager.java

```diff
@@ -0,0 +1,154 @@
+package com.minecolonies.coremod.colony.managers;
+
+import com.minecolonies.api.colony.IColony;
+import com.minecolonies.api.colony.managers.interfaces.IStatisticsManager;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.ListTag;
+import net.minecraft.nbt.Tag;
+import net.minecraft.network.FriendlyByteBuf;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+
+import static com.minecolonies.api.util.constant.NbtTagConstants.*;
+
+/**
+ * Manager for colony related statistics.
+ */
+public class StatisticsManager implements IStatisticsManager
+{
+    /**
+     * NBT tags.
+     */
+    private static final String TAG_STAT_MANAGER    = "stat_manager";
+    private static final String TAG_STAT            = "stat";
+
+    /**
+     * Colony reference
+     */
+    private final IColony colony;
+
+    /**
+     * The current stats of the colony.
+     */
+    private final Map<String, Map<Short, Integer>> stats = new HashMap<>();
+
+    /**
+     * Create a new stat manager.
+     * @param colony the colony to check.
+     */
+    public StatisticsManager(final IColony colony)
+    {
+        this.colony = colony;
+    }
+
+    @Override
+    public void increment(final @NotNull String id)
+    {
+        incrementBy(id, 1);
+    }
+
+    @Override
+    public void incrementBy(final @NotNull String id, int qty)
+    {
+        final Map<Short, Integer> innerMap = stats.computeIfAbsent(id, k -> new HashMap<>());
+        innerMap.put(colony.getDay(), innerMap.getOrDefault(colony.getDay(), 0) + qty);
```

Comment:
I'm not sure if I'm happy with the data structure, using a map of SHort,Integer means you always have to do unboxing for both values, which isnt really great performance wise, and increment should be a very fast operation.
I also feel we shouldnt keep track of all data, but just last 30? Wonder if we could use a limited size array as data structure then, or at least use Long/Short specialized collections to avoid so much unboxing

Reply:
Yeah, I wondered about this too, but I can't come up with anything more efficient. Could obviously insert a tuple, but then we'd have to do a lot more work on value read.

Reply:
Not sure how large those values get, but Integers <= 127 (and likely the same for Shorts) get cached, which saves the creation of objects at least.

Reply:
Not sure if they're loaded in-game or not, but there's a bunch of classes like `Short2LongSortedMap` in the JDK that try very hard to avoid boxing.

Or as someaddons suggested, you could use a `long[]` instead of a map, either as a circular buffer with "current day index" or shifting all the cells on day rollover so the current day is always `[0]`.  When a day falls off the end, you either discard it or you add it to a total rollup.

Another option is to use an enum key for your parent map instead of a string, which might be a little faster (and still supports stringification for serialization).

Reply:
Wanted to avoid enum keys for easy extension, I'll look into Short2IntOpenHashMap

Reply:
A sorted map might be more appropriate for this case, as I suggested.  Your day indexes are likely contiguous, it has methods to easily fetch a subrange of data, and if you ever want to do something like graph it you're going to want it to be ordered anyway.  (And if you don't want to do something like that at some point, then why separate it in the first place?)

Reply:
done

Reply:
sth